src/intersections/boxes.jl

```diff
@@ -34,3 +34,71 @@ function intersecttype(f::Function, b1::Box{Dim,T}, b2::Box{Dim,T}) where {Dim,T
     return FaceTouchingBoxes(Box(u, v)) |> f
   end
 end
+
+"""
+    intersecttype(function, ray, box)
+
+Calculate the intersection type of a `ray`` and a `box`` and apply function `f` to it.
+
+## References
+
+* Williams A, Barrus S, Morley R K, et al., 2005. [An efficient and robust ray-box
+  intersection algorithm]
+  (https://dl.acm.org/doi/abs/10.1145/1198555.1198748)
+"""
+function intersecttype(f::Function, r::Ray{3,T}, b::Box{3,T}) where {T}
+  invdir = one(T) ./ direction(r)
+  bounds = coordinates.(extrema(b))
+  orig = coordinates(origin(r))
+
+  # X slab
+  if invdir[1] ≥ 0
+    tmin = (bounds[1][1] - orig[1]) * invdir[1]
+    tmax = (bounds[2][1] - orig[1]) * invdir[1]
+  else
+    tmin = (bounds[2][1] - orig[1]) * invdir[1]
+    tmax = (bounds[1][1] - orig[1]) * invdir[1]
+  end
+  tmin = max.(tmin, zero(T))
```

Comment:
I don't think we need broadcasting here?

Reply:
Yes. We need not. That's a mistake.